gk/main.c

```diff
@@ -1437,6 +1439,66 @@ send_request_to_grantor(struct ipacket *packet, struct gk_fib *fib,
 		drop_packet_front(packet->pkt, instance);
 }
 
+static void
+lookup_fib_bulk(struct gk_lpm *ltbl, struct ip_flow **flows,
+	int num_flows, struct gk_fib *fibs[])
+{
+	int i;
+	/* The batch size for IPv4 LPM table lookup. */
+	const uint8_t FWDSTEP = 4;
+	const uint32_t default_nh = 0xFFFFFF;
+	int k = RTE_ALIGN_FLOOR(num_flows, FWDSTEP);
+
+	for (i = 0; i != k; i += FWDSTEP) {
+		int j;
+		const __m128i bswap_mask = _mm_set_epi8(12, 13, 14, 15, 8, 9, 10, 11,
+			4, 5, 6, 7, 0, 1, 2, 3);
+		__m128i dip = _mm_set_epi32(flows[i + 3]->f.v4.dst.s_addr,
+			flows[i + 2]->f.v4.dst.s_addr,
+			flows[i + 1]->f.v4.dst.s_addr,
+			flows[i]->f.v4.dst.s_addr);
+		rte_xmm_t dst;
+
+		/* Byte swap 4 IPV4 addresses. */
+		dip = _mm_shuffle_epi8(dip, bswap_mask);
+
+		rte_lpm_lookupx4(ltbl->lpm, dip, dst.u32, default_nh);
+
+		for (j = 0; j < FWDSTEP; j++) {
+			if (dst.u32[j] != default_nh)
+				fibs[i + j] = &ltbl->fib_tbl[dst.u32[j]];
+			else
+				fibs[i + j] = NULL;
+		}
+	}
+
+	for (i = k; i < num_flows; i++)
```

Comment:
1. Drop `i = k` above;
2. Add `RTE_VERIFY(i == k);` just before the `for` loop.